File: summer/WebContent/js/system/windows/controls/SelectionChangedEventArgs.js
/**
 * SelectionChangedEventArgs
 */

define(["dojo/_base/declare", "system/Type", "windows/RoutedEventArgs"], 
		function(declare, Type, RoutedEventArgs){
	var SelectionChangedEventArgs = declare("SelectionChangedEventArgs", RoutedEventArgs,{
		constructor:function(
	            /*RoutedEvent*/ id,
	            /*IList*/ removedItems, 
	            /*IList*/ addedItems)

        { 
			if(arguments.length == 3){
	            if (id == null) 
	                throw new ArgumentNullException("id");
	            if (removedItems == null) 
	                throw new ArgumentNullException("removedItems");
	            if (addedItems == null)
	                throw new ArgumentNullException("addedItems");
	 
	            this.RoutedEvent = id;
	 
	            this._removedItems = []; //new object[removedItems.Count]; 
	            removedItems.CopyTo(this._removedItems, 0);
	 
	            this._addedItems = []; //new object[addedItems.Count];
	            addedItems.CopyTo(this._addedItems, 0);
			}else{
				/*List<ItemsControl.ItemInfo>*/var unselectedInfos = id, /*List<ItemsControl.ItemInfo>*/ selectedInfos = removedItems;
		        { 
		            this.RoutedEvent = Selector.SelectionChangedEvent; 

		            this._removedItems = []; //new object[unselectedInfos.Count]; 
		            for (var i=0; i<unselectedInfos.Count; ++i)
		            {
		                this._removedItems[i] = unselectedInfos.Get(i).Item;
		            } 

		            this._addedItems = []; // new object[selectedInfos.Count]; 
		            for (var i=0; i<selectedInfos.Count; ++i) 
		            {
		                this._addedItems[i] = selectedInfos.Get(i).Item; 
		            }

		            this._removedInfos = unselectedInfos;
		            this._addedInfos = selectedInfos; 
		        }
			}
        },
		
        /// <summary> 
        /// This method is used to perform the proper type casting in order to 
        /// call the type-safe SelectionChangedEventHandler delegate for the SelectionChangedEvent event.
        /// </summary> 
        /// <param name="genericHandler">The handler to invoke.</param>
        /// <param name="genericTarget">The current object along the event's route.</param>
//        protected override void 
		InvokeEventHandler:function(/*Delegate*/ genericHandler, /*object*/ genericTarget)
        { 
            /*SelectionChangedEventHandler*/var handler = genericHandler;
 
            handler.Invoke(genericTarget, this); 
        }
	});
	
	Object.defineProperties(SelectionChangedEventArgs.prototype,{
		  
	});
	
	Object.defineProperties(SelectionChangedEventArgs,{

        /// <summary>
        /// An IList containing the items that were unselected during this event
        /// </summary> 
//        public IList 
		RemovedItems:
        { 
            get:function() { return this._removedItems; } 
        },
 
        /// <summary>
        /// An IList containing the items that were selected during this event
        /// </summary>
//        public IList 
        AddedItems: 
        {
            get:function() { return this._addedItems; } 
        }, 

        /// <summary> 
        /// A list containing the ItemInfos that were unselected during this event
        /// </summary> 
//        internal List<ItemsControl.ItemInfo> 
        RemovedInfos:
        {
            get:function() { return this._removedInfos; } 
        },

        /// <summary>
        /// A list containing the ItemInfos that were selected during this event 
        /// </summary>
//        internal List<ItemsControl.ItemInfo> 
        AddedInfos: 
        { 
            get:function() { return this._addedInfos; }
        } 	  
	});
	
	SelectionChangedEventArgs.Type = new Type("SelectionChangedEventArgs", 
			SelectionChangedEventArgs, [RoutedEventArgs.Type]);
	return SelectionChangedEventArgs;
});

//
//    /// <summary> 
//    /// The delegate type for handling a selection changed event
//    /// </summary>
//    public delegate void SelectionChangedEventHandler(
//        object sender, 
//        SelectionChangedEventArgs e);
//        private object[] _addedItems;
//        private object[] _removedItems; 
//        private List<ItemsControl.ItemInfo> _addedInfos; 
//        private List<ItemsControl.ItemInfo> _removedInfos;

File: org.summer.view.widget/src/org/summer/view/widget/controls/SelectionChangedEventArgs.java
package org.summer.view.widget.controls;

import org.summer.view.widget.ArgumentNullException;
import org.summer.view.widget.Delegate;
import org.summer.view.widget.RoutedEvent;
import org.summer.view.widget.RoutedEventArgs;
import org.summer.view.widget.collection.IList;
import org.summer.view.widget.collection.List;
import org.summer.view.widget.controls.primitives.Selector;
/// <summary> 
    /// The delegate type for handling a selection changed event
    /// </summary>
//    public delegate void SelectionChangedEventHandler(
//        Object sender, 
//        SelectionChangedEventArgs e);
 
    /// <summary> 
    /// The inputs to a selection changed event handler
    /// </summary> 
    public class SelectionChangedEventArgs extends RoutedEventArgs
    {
        #region Constructors
 
        /// <summary>
        /// The constructor for selection changed args 
        /// </summary> 
        /// <param name="id">The event ID for the event about to fire -- should probably be Selector.SelectionChangedEvent</param>
        /// <param name="removedItems">The items that were unselected during this event</param> 
        /// <param name="addedItems">The items that were selected during this event</param>
        public SelectionChangedEventArgs(
            RoutedEvent id,
            IList removedItems, 
            IList addedItems)
        { 
            if (id == null) 
                throw new ArgumentNullException("id");
            if (removedItems == null) 
                throw new ArgumentNullException("removedItems");
            if (addedItems == null)
                throw new ArgumentNullException("addedItems");
 
            RoutedEvent = id;
 
            _removedItems = new Object[removedItems.Count]; 
            removedItems.CopyTo(_removedItems, 0);
 
            _addedItems = new Object[addedItems.Count];
            addedItems.CopyTo(_addedItems, 0);
        }
 
        /*internal*/ public SelectionChangedEventArgs(List</*ItemsControl.*/ItemInfo> unselectedInfos, List</*ItemsControl.*/ItemInfo> selectedInfos)
        { 
            RoutedEvent = Selector.SelectionChangedEvent; 

            _removedItems = new Object[unselectedInfos.Count]; 
            for (int i=0; i<unselectedInfos.Count; ++i)
            {
                _removedItems[i] = unselectedInfos[i].Item;
            } 

            _addedItems = new Object[selectedInfos.Count]; 
            for (int i=0; i<selectedInfos.Count; ++i) 
            {
                _addedItems[i] = selectedInfos[i].Item; 
            }

            _removedInfos = unselectedInfos;
            _addedInfos = selectedInfos; 
        }
 
//        #endregion 

//        #region Public Properties 

        /// <summary>
        /// An IList containing the items that were unselected during this event
        /// </summary> 
        public IList RemovedItems
        { 
            get { return _removedItems; } 
        }
 
        /// <summary>
        /// An IList containing the items that were selected during this event
        /// </summary>
        public IList AddedItems 
        {
            get { return _addedItems; } 
        } 

//        #endregion 

//        #region Internal Properties

        /// <summary> 
        /// A list containing the ItemInfos that were unselected during this event
        /// </summary> 
        /*internal*/ public List</*ItemsControl.*/ItemInfo> RemovedInfos 
        {
            get { return _removedInfos; } 
        }

        /// <summary>
        /// A list containing the ItemInfos that were selected during this event 
        /// </summary>
        /*internal*/ public List</*ItemsControl.*/ItemInfo> AddedInfos 
        { 
            get { return _addedInfos; }
        } 

//        #endregion

//        #region Protected Methods 

        /// <summary> 
        /// This method is used to perform the proper type casting in order to 
        /// call the type-safe SelectionChangedEventHandler delegate for the SelectionChangedEvent event.
        /// </summary> 
        /// <param name="genericHandler">The handler to invoke.</param>
        /// <param name="genericTarget">The current Object along the event's route.</param>
        protected /*override*/ void InvokeEventHandler(Delegate genericHandler, Object genericTarget)
        { 
            SelectionChangedEventHandler handler = (SelectionChangedEventHandler)genericHandler;
 
            handler(genericTarget, this); 
        }
 
//        #endregion

//        #region Data
 
        private Object[] _addedItems;
        private Object[] _removedItems; 
        private List</*ItemsControl.*/ItemInfo> _addedInfos; 
        private List</*ItemsControl.*/ItemInfo> _removedInfos;
 
//        #endregion
    }